src/locations/locations.service.js

```diff
@@ -33,13 +38,19 @@ class LocationsService {
       options.query_params.state = state
     }
 
-    this.localDB = this.pouchDB('navIntLocationsDB')
-    this.replicationFrom = this.localDB.replicate.from(this.remoteDB, options)
+    if (!this.localDB) {
+      this.localDB = this.pouchDB('navIntLocationsDB')
+    }
 
-    Object.keys(this.onReplicationCompleteCallbacks)
-      .forEach((id) => registerCallback(this.replicationFrom, this.onReplicationCompleteCallbacks[id]))
+    if (!this.replicationFrom) {
+      this.replicationFrom = this.localDB.replicate.from(this.remoteDB, options)
+
+      Object.keys(this.onReplicationCompleteCallbacks)
+        .forEach((id) => registerCallback(this.replicationFrom, this.onReplicationCompleteCallbacks[id]))
+    }
 
     return this.replicationFrom
+      .catch(retry)
```

Comment:
This restarts the replication every time it fails until it completes. I was wondering if I should limit the number of retries, but it seems to me it can't do any harm by keeping trying?